crates/iceberg/src/table.rs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Table API for Apache Iceberg
+
+use crate::spec::TableMetadata;
+use std::collections::HashMap;
+
+/// Table represents a table in the catalog.
+pub struct Table {
+    metadata: TableMetadata,
+    location: String,
```

Comment:
I'm a bit obsessed by these things, but I prefer to keep the naming and order equal to the rest-api spec :)
```suggestion
    metadata_location: String,
    metadata: TableMetadata,
```